collect/collect.go

```diff
@@ -481,29 +519,51 @@ func (i *InMemCollector) redistributeTraces() {
 	}
 }
 
-func (i *InMemCollector) sendExpiredTracesInCache(now time.Time) {
-	traces := i.cache.TakeExpiredTraces(now)
+func (i *InMemCollector) sendExpiredTracesInCache(ctx context.Context, now time.Time) {
+	ctx, span := otelutil.StartSpan(ctx, i.Tracer, "sendExpiredTracesInCache")
+	defer span.End()
+
+	startTime := time.Now()
+	traces := i.cache.TakeExpiredTraces(now, int(i.Config.GetTracesConfig().MaxExpiredTraces))
+	dur := time.Now().Sub(startTime)
+
+	span.SetAttributes(attribute.Int("num_traces_to_expire", len(traces)), attribute.Int64("take_expired_traces_duration_ms", dur.Milliseconds()))
+
 	spanLimit := uint32(i.Config.GetTracesConfig().SpanLimit)
+
+	var totalSpansSent int64
+
 	for _, t := range traces {
+		totalSpansSent += int64(t.DescendantCount())
+		_, span2 := otelutil.StartSpanWith(ctx, i.Tracer, "sendReadyTrace", "num_spans", int64(t.DescendantCount()))
```

Comment:
Let's rename this span, something like `sendReadySpan`.

Reply:
This loop is working on each trace returned from the cache. That's why we named it with the `Trace` suffix instead of span